apps/gdalalg_raster_reproject.cpp

```diff
@@ -132,21 +132,71 @@ GDALRasterReprojectAlgorithm::GDALRasterReprojectAlgorithm(bool standaloneStep)
     AddArg("error-threshold", 0, _("Error threshold"), &m_errorThreshold)
         .AddAlias("et")
         .SetCategory(GAAC_ADVANCED);
+
+    AddNumThreadsArg(&m_numThreads, &m_numThreadsStr);
+}
+
+/************************************************************************/
+/*          GDALRasterReprojectAlgorithm::CanHandleNextStep()           */
+/************************************************************************/
+
+bool GDALRasterReprojectAlgorithm::CanHandleNextStep(
+    GDALRasterPipelineStepAlgorithm *poNextStep) const
+{
+    return poNextStep->GetName() == "write" &&
+           poNextStep->GetFormat() != "stream";
 }
 
 /************************************************************************/
 /*            GDALRasterReprojectAlgorithm::RunStep()                   */
 /************************************************************************/
 
-bool GDALRasterReprojectAlgorithm::RunStep(GDALProgressFunc, void *)
+bool GDALRasterReprojectAlgorithm::RunStep(
+    GDALRasterPipelineStepRunContext &ctxt)
 {
     CPLAssert(m_inputDataset.GetDatasetRef());
     CPLAssert(m_outputDataset.GetName().empty());
     CPLAssert(!m_outputDataset.GetDatasetRef());
 
     CPLStringList aosOptions;
-    aosOptions.AddString("-of");
-    aosOptions.AddString("VRT");
+    std::string outputFilename;
+    if (ctxt.m_poNextStep)
```

Comment:
see comment below about conveying the "can be handled" meaning within the pointer to the next step.